backend/btrixcloud/colls.py

```diff
@@ -77,75 +73,79 @@ async def add_collection(
+            raise HTTPException(status_code=400, detail="collection_name_taken")
 
-    async def update_collection(self, oid: uuid.UUID, name: str, update: UpdateColl):
+    async def update_collection(self, coll_id: uuid.UUID, update: UpdateColl):
         """Update collection"""
         query = update.dict(exclude_unset=True)
 
         if len(query) == 0:
             raise HTTPException(status_code=400, detail="no_update_data")
 
-        result = await self.collections.find_one_and_update(
-            {"name": name, "oid": oid},
-            {"$set": query},
-            return_document=pymongo.ReturnDocument.AFTER,
-        )
-        if not result:
-            raise HTTPException(status_code=404, detail="collection_not_found")
-
-        return result
+        try:
+            result = await self.collections.find_one_and_update(
+                {"_id": coll_id},
+                {"$set": query},
+                return_document=pymongo.ReturnDocument.AFTER,
+            )
+        except pymongo.errors.DuplicateKeyError:
+            # pylint: disable=raise-missing-from
+            raise HTTPException(status_code=400, detail="collection_name_taken")
 
-    async def rename_collection(self, oid: uuid.UUID, name: str, new_name: str):
-        """Rename collection"""
-        result = await self.collections.find_one_and_update(
-            {"name": name, "oid": oid},
-            {"$set": {"name": new_name}},
-        )
         if not result:
             raise HTTPException(status_code=404, detail="collection_not_found")
 
-        return new_name
+        result["id"] = result["_id"]
+        del result["_id"]
 
-    async def add_crawl_to_collection(self, oid: uuid.UUID, name: str, crawl_id: str):
+        return result
+
+    async def add_crawl_to_collection(self, coll_id: uuid.UUID, crawl_id: str):
         """Add crawl to collection"""
         result = await self.collections.find_one_and_update(
-            {"name": name, "oid": oid},
+            {"_id": coll_id},
             {"$push": {"crawlIds": crawl_id}},
             return_document=pymongo.ReturnDocument.AFTER,
         )
         if not result:
             raise HTTPException(status_code=404, detail="collection_not_found")
+
+        result["id"] = result["_id"]
+        del result["_id"]
+
         return result
 
-    async def remove_crawl_from_collection(
-        self, oid: uuid.UUID, name: str, crawl_id: str
-    ):
+    async def remove_crawl_from_collection(self, coll_id: uuid.UUID, crawl_id: str):
         """Remove crawl from collection"""
         result = await self.collections.find_one_and_update(
-            {"name": name, "oid": oid},
+            {"_id": coll_id},
             {"$pull": {"crawlIds": crawl_id}},
             return_document=pymongo.ReturnDocument.AFTER,
         )
         if not result:
             raise HTTPException(status_code=404, detail="collection_not_found")
+
+        result["id"] = result["_id"]
```

Comment:
If we want to be consistent, should probably do `Collection.from_dict()` here, and then return Collection as the object and have it be set as the response model..